File: Задание2/prog2.cs
using System;

class Program
{
    static void Main()
    {
        int countMatch = 0;
        int nextRound = int.Parse(Console.ReadLine());

        while (nextRound > 1)
        {
            if (nextRound % 2 == 1)
            {
                nextRound = (nextRound - 1) / 2 + 1;
                countMatch += (nextRound - 1);
            }
            else
            {
                nextRound /= 2;
                countMatch += nextRound;
            }
        }

        Console.WriteLine(countMatch);
    }
}

File: Задание2/prog2.rs
use std::io::{self, BufRead};

fn main() {
    let mut count_match = 0;
    let mut next_round = String::new();

    io::stdin().lock().read_line(&mut next_round).unwrap();
    let mut next_round: i32 = next_round.trim().parse().unwrap();

    while next_round > 1 {
        if next_round % 2 == 1 {
            next_round = (next_round - 1) / 2 + 1;
            count_match += next_round - 1;
        } else {
            next_round /= 2;
            count_match += next_round;
        }
    }

    println!("{}", count_match);
}
